azurerm/internal/services/network/virtual_network_gateway_connection_resource.go

```diff
@@ -455,6 +461,10 @@ func resourceVirtualNetworkGatewayConnectionDelete(d *schema.ResourceData, meta
 		return err
 	}
 
+	// We do not know the gateway name here, so have to lock on a fake name
```

Comment:
could we not lock on the ID or part of it instead?

Reply:
I think we could, but what would be the difference/benefit here?